### DISCUSSION TITLE
Feature Request: Instead of just exposing the crud endpoints, can it generate the actual methods in a file?

### DISCUSSION THREAD

**Discussion by arun-pt** [piccolo-orm/piccolo_api#314] [Ideas]
Idea is to utilize piccolo FastAPIWrapper to generate both router files and service/repository files for the basic crud actions on a table.
Reference: https://docs.nestjs.com/recipes/crud-generator

Advantages:
1. It can then be easy to override logic based on one's use cases.
2. The generated method can then be called from other services directly instead of relying only on a HTTP call
3. Useful to scaffold a larger fastapi project with minimal effort.

BTW, I have been already enjoying working with piccolo with its many awesome features. Thanks a lot.

**Comment by dantownsend** [comment]
It's an interesting idea.

We currently have some code scaffolding available via `piccolo asgi new`, but this is just for a new project.

We also have `piccolo app new` which generates a new Piccolo app, but doesn't contain any endpoints.

If we were to add more scaffolding, it would probably be augmenting these existing commands.

For example `piccolo asgi route`.

**Comment by ghost** [comment]
`piccolo asgi route` sounds great.
